### ISSUE TITLE
[HOLD for payment 2024-08-29] [$250] mWeb/Safari - Keypad overlap composer & blank space in header when open admins room from WS share pg

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#47063]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.18-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y 
**If this was caught during regression testing, add the test name, ID and link from TestRail:** Exploratory around https://github.com/Expensify/App/pull/46855
**Email or phone of affected tester (no customers):** applausetester+1088ck@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Go to Settings >> Workspaces >> Select any workspace (or create a new one)
3. Open Profile >> Share option
4. Tap #admins text link
5. Tap on composer (Note: Keypad overlaps composer)
6. Start typing any message

## Expected Result:
1. Keypad should not overlap on composer & user should be able to view the message entered in composer
2. No blank space displayed in header

## Actual Result:
1. Keypad overlaps composer & user not able to view the message entered in composer
2. Blank space displayed in header

## Workaround:
Unknown

## Platforms:

- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/cc9434b2-4622-49cb-86a4-f071a3f10d2e

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~011db3b268b189dcbe</li>
        <li>Upwork Job ID: 1822989061968230556</li>
        <li>Last Price Increase: 2024-08-12</li>
<li>Automatic offers: </li>
<ul>
<li>jjcoffee | Reviewer | 103560629</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mallenexpensify</details>

**Comment by IuliiaHerets** [comment]
We think that this bug might be related to #vip-vsb

**Comment by IuliiaHerets** [comment]
@zanyrenney  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by zanyrenney** [comment]
polish but worth fixing due to the appearance/functionality.

**Comment by zanyrenney** [comment]
adding external.

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Blank space and keyboard overlap the composer when focus on composer on iOS mWeb safari.

### What is the root cause of that problem?
When we focus on the composer, the page is scrolled to the top so the composer is positioned above the keyboard and a margin-top is applied so the report header is shown at the top.
https://github.com/Expensify/App/blob/b7f7f1b4f36451b4e3a763ac717e1c8fd645ee5a/src/pages/home/ReportScreen.tsx#L280-L283

For webkkit based browsers, we have a hook to correct the visual viewport scroll if `enable` is true. It's basically to "prevent" the user from scrolling the visual viewport.
https://github.com/Expensify/App/blob/b7f7f1b4f36451b4e3a763ac717e1c8fd645ee5a/src/hooks/useTackInputFocus/index.ts#L31-L53

`enable` is true if `shouldEnableMaxHeight` and `shouldAvoidScrollOnVirtualViewport` are true and it will register a visual viewport scroll listener.
https://github.com/Expensify/App/blob/b7f7f1b4f36451b4e3a763ac717e1c8fd645ee5a/src/components/ScreenWrapper.tsx#L222

When the report screen viewport is scrolled so that the composer is shown at the top of the keyboard, the scroll listener is triggered which corrects the scroll position back to 0.

But the problem here is, `ReportScreen`'s `ScreenWrapper` doesn't enable `shouldEnableMaxHeight`, so the hook should be disabled, but somehow the listener is still registered and triggered.
https://github.com/Expensify/App/blob/b7f7f1b4f36451b4e3a763ac717e1c8fd645ee5a/src/pages/home/ReportScreen.tsx#L733-L738

That's because when we open the workspace profile page > share > press `#admins`, the profile page is wrapped with WorkspacePageWithSections which enables `shouldEnableMaxHeight`.
https://github.com/Expensify/App/blob/b7f7f1b4f36451b4e3a763ac717e1c8fd645ee5a/src/pages/workspace/WorkspacePageWithSections.tsx#L171-L174

So, in short, the profile page screen wrapper settings affect the report screen.

### What changes do you think we should make in order to solve the problem?

Disable the hook if the screen isn't focused.
```
const isFocused = useIsFocused();
const isAvoidingViewportScroll = useTackInputFocus(isFocused && shouldEnableMaxHeight && shouldAvoidScrollOnVirtualViewport && Browser.isMobileWebKit());
```

**Comment by zanyrenney** [comment]
@jjcoffee please can you review the proposal above. It seems like a UI issue as opposed to anything critical on the functionality. Should be a simple enough!

**Comment by jjcoffee** [comment]
@bernhardoj's [proposal](https://github.com/Expensify/App/issues/47063#issuecomment-2287872529) LGTM! (Really well written too, which is always a bonus!)

:ribbon::eyes::ribbon: C+ reviewed

**Comment by bernhardoj** [comment]
PR is ready

cc: @jjcoffee

**Comment by zanyrenney** [comment]
reapplying bug label for payout as I am going to be OOO until next week so will miss this payment date. making daily as payout in 3 days! thanks so much @mallenexpensify

**Comment by jjcoffee** [comment]
- The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/23185
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/commit/a32fa95627b8a5a9a093c8e05e79dd2facf9f8a9#r145836648
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A - broader testing required
- Determine if we should create a regression test for this bug. **Yes**
- If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

**Regression Test Proposal**

iOS Safari only

1. Go to Settings >> Workspaces >> Select any workspace (or create a new one)
2. Open Profile >> Share option
3. Press the #admins text link
4. Focus on the composer
5. Verify the header still sits at the top and the composer is above the keyboard

Do we agree 👍 or 👎

**Comment by zanyrenney** [comment]
payment didn't happen in my absence so handling it now as the initial assignee :)

**Comment by zanyrenney** [comment]
**payment summary**

paid @jjcoffee  $250 via Upwork
@bernhardoj is due $250 and is paid via ND payments.

**Comment by JmillsExpensify** [comment]
$250 approved for @bernhardoj
